src/porepy/models/contact_mechanics_biot_model.py

```diff
@@ -780,19 +780,28 @@ def _assign_discretizations(self) -> None:
                 * accumulation_primary
                 + subdomain_proj_scalar.cell_prolongation(g_frac) * accumulation_fracs
             )
-
-            interface_flow_eq = robin_ad.mortar_scaling * (
-                mortar_proj_scalar.primary_to_mortar_avg
+            # Interface equation: \lambda = -\kappa (p_l - p_h)
+            # Robin_ad.mortar_discr represents -\kappa. The involved term is
+            # reconstruction of p_h on internal boundary, which has contributions
+            # from cell center pressure, external boundary and interface flux
+            # on internal boundaries (including those corresponding to "other"
+            # fractures).
+            interface_flow_eq = (
+                robin_ad.mortar_discr
+                * mortar_proj_scalar.primary_to_mortar_avg
                 * mpfa_ad.bound_pressure_cell
                 * p
-                + mortar_proj_scalar.primary_to_mortar_avg
+                + robin_ad.mortar_discr
+                * mortar_proj_scalar.primary_to_mortar_avg
                 * mpfa_ad.bound_pressure_face
-                * (
-                    mortar_proj_scalar.mortar_to_primary_int * mortar_flux
-                    + bc_val_scalar
-                )
-                - mortar_proj_scalar.secondary_to_mortar_avg * p
-                + robin_ad.mortar_discr * mortar_flux
+                * mortar_proj_scalar.mortar_to_primary_int
+                * mortar_flux
+                + robin_ad.mortar_discr
+                * mortar_proj_scalar.primary_to_mortar_avg
+                * mpfa_ad.bound_pressure_face
+                * bc_val_scalar
+                - robin_ad.mortar_discr * mortar_proj_scalar.secondary_to_mortar_avg * p
+                + mortar_flux
```

Comment:
Can we collect terms and use parentheses here?

Reply:
Probably. I'll see if ad allows it